napari/components/add_layers_mixin.py

```diff
@@ -45,6 +50,57 @@ def add_layer(self, layer):
         if len(self.layers) == 1:
             self.reset_view()
 
+    def add_path(
+        self,
+        path: Union[str, Sequence[str]],
+        stack: bool = False,
+        use_dask: Optional[bool] = None,
+    ) -> List[layers.Layer]:
+        """Add a path to the viewer
+
+        Paths will be handed one-by-one to the napari_get_reader hook.
+
+        Parameters
+        ----------
+        path : str or list of str
+            path(s) to view.
+        stack : bool, optional
+            Concatenate multiple input files into a single stack,
+            by default False
+        use_dask : bool, optional
+            Whether to use dask to create a lazy array, rather than NumPy.
+            If ``None``, will resolve to True if filenames contains more than
+            one image, False otherwise.  by default None.
+        """
+        paths = [path] if isinstance(path, str) else path
+        if not isinstance(paths, (tuple, list)):
+            raise ValueError(
+                "'path' argument must be a string, list, or tuple"
+            )
+
+        added_layers: List[layers.Layer] = []
+        if stack:
+            images = io.magic_imread(paths, use_dask=use_dask, stack=stack)
+            added = self.add_image(images)
+            added = added if isinstance(added, list) else [added]
+            added_layers.extend(added)
+            return added
+
+        # iterate through each path provided, looking for a suitable reader
```

Comment:
Two possibilities:

(1) `get_reader` can also accept a list of paths, and return a plugin that can read them all, or
(2) plugin returned images can also be stacked.

I kinda like the first one better.